var/spack/repos/builtin/packages/py-scikit-build-core/package.py

```diff
@@ -51,6 +77,8 @@ class PyScikitBuildCore(PythonPackage):
     depends_on("py-setuptools", type="test")
     depends_on("py-wheel", type="test")
 
+    depends_on("py-importlib-metadata", type=("build", "run"), when="^python@:3.7")
```

Comment:
Only remaining comment from my side is to move this up to the dependencies block so we keep the dependencies lexically sorted.

Reply:
Should be done